perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/RegistrarManagerImpl.java

```diff
@@ -3716,6 +3801,65 @@ public void addGroupsToAutoRegistration(PerunSession sess, List<Group> groups) t
 		perun.getGroupsManagerBl().addGroupsToAutoRegistration(sess, groups);
 	}
 
+	@Override
+	public void addGroupsToAutoRegistration(PerunSession sess, List<Group> groups, ApplicationFormItem formItem) throws GroupNotExistsException, PrivilegeException, GroupNotAllowedToAutoRegistrationException {
+		Utils.checkPerunSession(sess);
+
+		for (Group group : groups) {
+			perun.getGroupsManagerBl().checkGroupExists(sess, group);
+		}
+
+		//Authorization
+		for (Group group : groups) {
+			if (!AuthzResolver.authorizedInternal(sess, "addGroupsToAutoRegistration_List<Group>_ApplicationFormItem_policy", group)) {
+				throw new PrivilegeException(sess, "addGroupsToAutoRegistration");
+			}
+		}
+
+		// Create application form if non exists
```

Comment:
There are a few lines of code in submitEmbeddedGroupApplications() that also create a new form if it does not exist, which are obsolete now (lines 1909-1913 I believe)

Reply:
Removed from msubmitEmbeddedGroupApplications().